src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -133,19 +143,26 @@ private IEnumerable<string> GetSearchUrlsForPolling(string originalBaseUrl)
             }
         }
 
-        public async Task<IReadOnlyList<string>> GetSearchBaseUrlsAsync()
+        public async Task<IReadOnlyList<SearchService>> GetSearchServicesAsync()
         {
-            var searchBaseUrls = new List<string>();
-            if (_testSettings.SearchBaseUrl != null)
+            if (_azureManagementAPIWrapper != null)
             {
-                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+                string result = await _azureManagementAPIWrapper.GetCloudServicePropertiesAsync(
+                                    _testSettings.Subscription,
+                                    _testSettings.SearchServiceResourceGroup,
+                                    _testSettings.SearchServiceName,
+                                    _testSettings.SearchServiceSlot,
+                                    CancellationToken.None);
+
+                var cloudService = AzureHelper.ParseCloudServiceProperties(result);
+
+                return new List<SearchService>() { new SearchService { Uri = cloudService.Uri, InstanceCount = cloudService.InstanceCount } };
             }
             else
             {
-                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrlsAsync());
+                var urls = await _v3IndexClient.GetSearchBaseUrlsAsync();
+                return urls.Select(url => new SearchService { Uri = new Uri(url), InstanceCount = DefaultServiceInstanceCount }).ToList();
```

Comment:
Should test output indicate whether AzureManagement API was used versus the default?